### PR TITLE
Arduino as an ESP32 component (VSC-354)

### PR DESCRIPTION
**Pull request by brianignacio5**
Introduce cloning [Arduino ESP32](https://github.com/espressif/arduino-esp32) as ESP-IDF component in two ways:

1) **Create ESP-IDF project** with the `arduino-as-component` will create a new project with Arduino as component and define a template `main.cpp`
2) Will just clone Arduino ESP32 as component for the currently selected workspace folder.

Added choosing directory for **Create ESP-IDF project** with validations.

### CODE REVIEW

File: src/espIdf/arduino/addArduinoComponent.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  private async checkIdfVersion(espIdfPath?: string) {
+    if (typeof espIdfPath === "undefined" || espIdfPath === "") {
+      espIdfPath =
+        idfConf.readParameter("idf.espIdfPath") || process.env.IDF_PATH;
+    }
+    const idfVersion = await getEspIdfVersion(espIdfPath);
+    const results = {
+      "4.0": "idf-release/v4.0",
+      "3.3": "idf-release/v3.3",
```
Comment (pwmb) [inline src/espIdf/arduino/addArduinoComponent.ts:89]: only this two specific versions are allowed ?

File: src/espIdf/arduino/addArduinoComponent.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      OutputChannel.appendLine("\n❌ [Arduino ESP32 Cloning] : Stopped!\n");
+    }
+  }
+
+  public async cloneArduinoInComponentsFolder(branchToUse: string) {
+    const componentsDir = join(this.projectDir, "components");
+    await ensureDir(componentsDir);
+    return new Promise((resolve, reject) => {
+      this.cloneProcess = spawn(
+        `git`,
```
Comment (pwmb) [inline src/espIdf/arduino/addArduinoComponent.ts:50]: Maybe call `checkGitExists()` function before this to ensure user had git installed and if not show the specific error message (just a precautionary step)

File: src/extension.ts [hunk 3]
```diff
@@ -255,22 +256,121 @@ export async function activate(context: vscode.ExtensionContext) {
...
     });
   });
 
+  registerIDFCommand("espIdf.addArduinoAsComponentToCurFolder", () => {
+    PreCheck.perform([openFolderCheck], () => {
+      vscode.window.withProgress(
+        {
+          cancellable: true,
+          location: vscode.ProgressLocation.Notification,
+          title: "ESP-IDF:",
```
Comment (pwmb) [inline src/extension.ts:353]: Title is very generic

File: src/espIdf/arduino/addArduinoComponent.ts (hunk 1 above)
Comment (brianignacio5) [reply src/espIdf/arduino/addArduinoComponent.ts:89]: The Arduino ESP32 as component doesn't work with other versions I tested (v4.1, v4.2)

File: src/espIdf/arduino/addArduinoComponent.ts (hunk 2 above)
Comment (brianignacio5) [reply src/espIdf/arduino/addArduinoComponent.ts:50]: `getEspIdfVersion` calls `checkGitExists`

**Comment by pwmb** [comment]
@brianignacio5 you want to release this in `v0.3.0` or `v0.4.0`; if `v0.4.0` please don't merge it now, wait for `0.3.0` to be release next week

**Pull request merged**